### DISCUSSION TITLE
set transformCSS to 'pre' still unable to use @apply

### DISCUSSION THREAD

**Discussion by Garfield550** [unocss/unocss#1187] [Q&A] [answered]
Vite + Vue + UnoCSS

```json
{
  "dependencies": {
    "@unocss/reset": "^0.41.1",
    "vue": "^3.2.37"
  },
  "devDependencies": {
    "@unocss/preset-web-fonts": "^0.41.1",
    "@vitejs/plugin-vue": "^2.3.3",
    "unocss": "^0.41.1",
    "vite": "^2.9.12"
  }
}
```

vite.config.ts

```typescript
export default defineConfig({
  plugins: [
    vue(),
    Unocss({
      presets: [
        presetUno(),
        presetWebFonts({
          provider: 'google',
          fonts: { sans: 'Montserrat' }
        })
      ],
      transformCSS: 'pre'
    })
  ]
})
```

base.css

```css
html {
  @apply font-sans;
}
```

**Comment by Garfield550** [comment, marked as answer]
Found solution from issue #649 
I need to use `@unocss/transformer-directives`, this transformer does not built in.

```ts
export default defineConfig({
  plugins: [
    Unocss({
      transformers: [transformerDirective()]
    })
  ]
})
```

**Reply by clouds56** [reply]
This make `@apply` works in vite, but vscode still gives me a warning `Unknown at rule @apply css(unknownAtRules)`.

**Reply by zyyv** [reply]
replace to `--uno` `--at-apply` https://unocss.dev/transformers/directives#at-apply

**Comment by zyyv** [comment]
I'm not found any apply about `transformCSS` in source code. And i found you resolved it.😄

**Reply by Garfield550** [reply]
I found a type declare in https://github.com/unocss/unocss/blob/4d5a18e3c35f64d041a27dfe9a3668d47a38e5ae/packages/vite/src/types.ts#L30
It misled me :(
